### DISCUSSION TITLE
linuxbrew and xorg issues

### DISCUSSION THREAD

**Discussion by lteneyck** [Homebrew/discussions#351] [Linux]
I am running into problems with warning about deprecated xorg routines. I note that there are non-xorg versions of them, apparently using the Wayland protocol instead of X11. This is well and good, but during this transition I am finding that when I duly update, unlink, uninstall, clean, and update again, (after untapping linuxbrew/xorg, major packages I need, like qt and mesa, immediately re-tap xorg and install the deprecated versions of these various libraries.

Is there something simple I can do to fix this? I know enough to recognize the problems and perhaps identify the sources, but I am not sure about how to fix any of these (if they require formula fixes), but I might be able to help.

Frankly the compatibility issues you guys face with divergent systems, versions, compilers, etc. is mind-boggling and I take my hat off to you. Homebrew is a major help for me. I have been fighting non-portability and evolving systems and standards for over fifty years and this is the most coherent assault on these problems I have seen in all that time.

**Comment by jonchang** [comment]
These deprecation notices will stop very soon as we are almost done migrating all necessary formulae from linuxbrew/xorg into the main homebrew/core tap. In the meantime, it's possible that an older formula you have installed (possibly from a third-party tap) still refers to linuxbrew/xorg, so it's worth it to do a quick check if that's the case. 

```
cd $(brew --cellar)
find . -name .brew -type d | xargs grep -R linuxbrew/xorg
```

**Reply by lteneyck** [reply]
Thank you! The results are mesa 20.3.1_3, which depends on libomxil-bellagio, libva, and libvdpau from xorg,
and xcb-util-image, which depends on xcb-util from xorg. This in turn means that anything that pulls in mesa, which is virtually everything that uses GL, will still try to pull in xorg.

The good news is that nearly everything I was trying to build traces back to this one formula.
Thanks again,
Lynn Ten Eyck

**Reply by iMichka** [reply]
We are still finishing a big migration. There are still 2 formulae to move from xorg to homebrew-core. Once this is done the xorg repo will not be needed anymore. We will need a few more days to finish this.

**Reply by mlavi** [reply]
@iMichka Anyplace I watch for the xorg migration conclusion? Thank you.

**Reply by iMichka** [reply]
See https://github.com/Homebrew/brew/issues/7028, it's one of the steps of that issue. The xorg migration is done.
